File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {

  if (input === input.toUpperCase() && input.match(/[A-Z]/)) {
    return 'Whoa, chill out!'
  } else if (input.slice(-1) === '?') {
    return 'Sure.'
  } else if (!input.replace(/^\s+|\s+$/g,"")) {
    return 'Fine. Be that way!'
  } else return 'Whatever.'
};

module.exports = Bob;

File: ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    if remark.upcase == remark && remark =~ /[A-Z]/
      'Whoa, chill out!'
    elsif remark.end_with? '?'
      'Sure.'
    elsif remark.strip.empty?
      'Fine. Be that way!'
    else
      'Whatever.'
    end
  end
end
